src/AppInstallerCLICore/Workflows/MSStoreInstallerHandler.cpp

```diff
@@ -111,7 +125,7 @@ namespace AppInstaller::CLI::Workflow
         AppInstallManager installManager;
 
         // Verifying/Acquiring product ownership
-        if (!GetFreeUserEntitlement(context, productId))
+        if (!GetFreeUserEntitlement(context, productId) || !GetFreeDeviceEntitlement(context, productId))

```

Comment:
This condition is not what we want.  For instance, it calls `GetFreeDeviceEntitlement` if `GetFreeUserEntitlement` returns true (aka, getting a user entitlement succeeded), and not if it returns false.  You wanted `&&`.

Creating a single `GetFreeEntitlement` as suggested above though would make the call site less complex, and is really needed to prevent the output messages from being duplicated in the event of errors.